### DISCUSSION TITLE
argo-workflows/workflow-archive/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#8935] [General]
# argo-workflows/workflow-archive/

https://argoproj.github.io/argo-workflows/workflow-archive/

**Comment by KilJaeeun** [comment]
Hello, I am a developer studying argo.
I don't know why I have to archive workflow. 

Do you store workflow information in a database such as mysql, not a workflow on Kubernetes resources?

**Reply by DiegoFlop** [reply]
On database only keep the result of wokflows, don't keep outpus, logs, and anything other data.
If you need keep outputs of pods, you must activate `archiveLogs: true` and save that on bucket S3 for example.

**Comment by MandeepAtLark** [comment]
Hello, My scenario is I have an old workflow before enabling workflow archive and saving the workflows in the database. If I patch my old workflow resources with `ttlStrategy` to archive them. After the workflow is removed from UI, will the workflow end up in my database? Thank you!

**Comment by S0PEX** [comment]
Hey,
I am also currently checking out argo to automate ml workflows.
As far as I can tell you can easily submit the .yml workflow using the ArgoCli. Is there also an easy way to submit them through a api endpoint? Or are you aware of any Couler alternative for NodeJs?

Regards,
Artur

**Reply by jwylliem** [reply]
there is a swagger API that you can build into node client.

**Comment by Amirioelmos** [comment]
hi there,
what the difference between argo cd and argo work flow?

**Reply by Amirioelmos** [reply]
i find this:
Argo CD (GitOps controller)
Argo Rollouts (Progressive Delivery controller)
Argo Workflows (Workflow engine for Kubernetes)
Argo Events (Event handling for Kubernetes)

**Reply by makzzz1986** [reply]
In simple words:
* Argo CD provides visual representation of Kubectl
* Argo Rollouts provides better ways of deployment strategies
* Argo Workflows is good for on-time jobs, cronjobs, pipelines (like Airflow) for ML

**Reply by timofey-drozhzhin** [reply]
CI and CD are the most common types of pipelines used to deploy code to the server.

- Argo CD is basically CD (Continuous Deployment). It syncs your repo to your your server. It doesn't exactly mirror the files over, instead it translates the instructions into actions. That way you don't have to SSH inside your server, or manually create Kubernetes resources (aka imperative programming). You specify all settings in your YAML manifest (aka declarative programming) and have Argo CD automatically render the infrastructure to match your declarations.
- Argo Workflows is mostly CI (Continuous Integration). It basically runs custom scripts, i.e. compiles the code before it's 
deployed, triggers backups, updates, etc. It does everything CD cannot. To be more precise, Argo Workflows sits above CI & CD, since it has control over CD. There's also Argo Events that plugs into Argo Workflows to add sensors.

PS - I'm new-ish to this. Open to criticism :)

**Comment by makzzz1986** [comment]
Is it possible to move migration as a separated parameter of workflow-controller to make an init container for Mysql deployment?

**Comment by shivam01anand** [comment]
I have a basic understanding issue ->

Argo workflow is a resource (albeit custom) which uses another resource i.e Pods to run workflows. 

When we use 
kubectl apply -n argo -f https://raw.githubusercontent.com/argoproj/argo-workflows/master/manifests/quick-start-postgres.yaml
we are building this resource using multiple immutable configs in the yaml file, similar to installing a software using exe?

Where then is the concept of operator used? 

Also, to run argo CLI we use another yaml file with dags etc which has a different format to yaml given to kubectl apply? 

Argo takes this new type of yaml and uses another resource i.e pods to run the containers etc.?

OR 

kubectl apply argo... thats the CRD and the new type of yaml file is the Controller..i.e declared final state needed.... these two constitute the "operator" which is invisible, but formed of these two?

**Comment by easylab2007** [comment]
if I want to keep workflow records of argo web for 30 days, then how can I set this?
